### ISSUE TITLE
Training a new PLC model

### ISSUE THREAD

**Issue by dariadiatlova** [xiph/LPCNet#205]
Hello, 

I am trying to train a new PLC model with [train_plc.py](https://github.com/xiph/LPCNet/tree/master/training_tf2/train_plc.py) and and several things have caused me difficulties:
1) Can I follow the same data preprocessing set up with `./dump_data -train input.s16 features.f32 data.s16` to get input [features.f32](LPCNet/blob/master/training_tf2/train_plc.py#L36) for PLC model training? 
2) How should [lost_file](https://github.com/xiph/LPCNet/tree/master/training_tf2/train_plc.py#L37) for training look like? Is it a single `.txt' file - a concatenation of smaller .txt files with one entry per 20ms packet, where 1 means "packet lost" and 0 means "packet not lost"? How to create a single file if original data was augmented after running` ./dump_data?` Is there any script for it? 
3) To close the above questions with lost_file preprocessing, can I just uncomment [the line](https://github.com/xiph/LPCNet/tree/master/training_tf2/plc_loader.py#L50) and train the model with random packets marked as lost? Have you noticed any significant degradation in how this works?
4) Following `test_plc.py`, the output is: [features + (1-lost)*out](https://github.com/xiph/LPCNet/tree/master/training_tf2/test_plc.py#L90), but the shapes:
- features: [bs, seq_len, nb_used_features+nb_burg_features]
- lost: [bs,  seq_len, 1]
- out: [bs,  nb_used_features, 1]

Did I think of the wrong shapes? What should be the shape of a correct output for writing to `output.f32`?

Thank you for sharing your code and supporting this repository!

**Comment by Janne-byti** [comment]
@dariadiatlova Hello, could you please retrain the PLC model? May I ask how to prepare training data?

**Comment by felixshing** [comment]
Hello, I am also interested in training a new PLC model with a combination of audio features and visual features. I would like to ask how to do that? Any insights are truly appreciated!

**Comment by YYX666660** [comment]
About the [lost_file](https://github.com/xiph/LPCNet/tree/master/training_tf2/train_plc.py#L37), I try to use the [lossgen](https://gitlab.xiph.org/xiph/opus/-/tree/main/dnn/torch/lossgen) model to generate the txt, but I'm not sure how long and how many loss rate should be generated. 
Very confused about training the PLC model...
